src/Dialogs/FrequencyDialog.cpp

```diff
@@ -0,0 +1,199 @@
+  : public ListWidget {
+
+  const DialogLook &dialog_look;
+  TextRowRenderer row_renderer;
+  Button *active_button, *standby_button, *cancel_button;
+  std::vector<RadioChannel> *channels;
+
+public:
+  void CreateButtons(WidgetDialog &dialog);
+
+public:
+  FrequencyListWidget(const DialogLook &_dialog_look, std::vector<RadioChannel> *_channels)
+    :dialog_look(_dialog_look),
+	 channels (_channels) {}
+
+  bool UpdateList() noexcept;
+
+  unsigned GetCursorIndex() const {
+    return GetList().GetCursorIndex();
+  }
+
+public:
+  /* virtual methods from class Widget */
+  void Prepare(ContainerWindow &parent, const PixelRect &rc) noexcept override;
+
+  /* virtual methods from class List::Handler */
+  void OnPaintItem(Canvas &canvas, const PixelRect rc,
+                   unsigned index) noexcept override {
+    assert(index < channels->size());
+
+    const RadioChannel &channel = channels->at(index);
+
+    FrequencyListRenderer::Draw(canvas, rc, channel,
+	                               row_renderer);
+  }
+
+  bool CanActivateItem(unsigned index) const noexcept override {
+    return true;
+  }
+
+  void OnActivateItem(unsigned index) noexcept override;
+};
+
+void
+FrequencyListWidget::CreateButtons(WidgetDialog &dialog)
+{
+	  standby_button = dialog.AddButton(_("Set Standby Frequency"), [this](){
+	    unsigned index = GetCursorIndex();
+	    assert(index < channels->size());
+	    const RadioChannel *channel = &channels->at(index);
+	    ActionInterface::SetStandbyFrequency(channel->radio_frequency,
+	            channel->name.c_str());
+	    cancel_button->Click();
+});
+
+	  active_button = dialog.AddButton(_("Set Active Frequency"), [this](){
+	    unsigned index = GetCursorIndex();
+	    assert(index < channels->size());
+	    const RadioChannel *channel = &channels->at(index);
+	    ActionInterface::SetActiveFrequency(channel->radio_frequency,
+	            channel->name.c_str());
+	    cancel_button->Click();
+  });
+
+  cancel_button = dialog.AddButton(_("Close"), mrCancel);
+}
+
+void
+FrequencyListWidget::Prepare(ContainerWindow &parent,
+                              const PixelRect &rc) noexcept
+{
+  CreateList(parent, dialog_look, rc,
+             row_renderer.CalculateLayout(*dialog_look.list.font_bold));
+
+  GetList().SetLength(channels->size());
+}
+
+void
+FrequencyListWidget::OnActivateItem(unsigned index) noexcept
+{
+  // May still be talking on active frequency
+  standby_button->Click();
+}
+
+bool
+FrequencyListWidget::UpdateList() noexcept
+{
+  channels->clear();
+
+  auto path = Profile::GetPath(ProfileKeys::FrequenciesFile);
+  if (path == nullptr) {
+    return false;
+  }
+
+  FileLineReader reader(path, Charset::AUTO);
+
+  TCHAR *line;
+  while ((line = reader.ReadLine()) != NULL) {
+
+	if (StringIsEmpty(line))
+	  continue;
+
+	TCHAR ctemp[4096];
+	if (_tcslen(line) >= ARRAY_SIZE(ctemp))
+	  /* line too long for buffer */
+	  continue;
+
+	const TCHAR *params[20];
```

Comment:
Why ... 20?

Reply:
Misunderstanding of how ExtractParameters behaves when parsing too many parameters; now fixed.